### PR TITLE
Contexts edge cases3

### PR DESCRIPTION
**Pull request by pmuir**

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 485fdffc8cbb641abbd0e6ef7509c529e82545ec
  * Your commit message does not contain a body

 * Commit: ebd507e1c61852b8d360aab858ab97473287d1bd
  * Your commit message does not contain a body

Guidelines are available at https://github.com/fabric8io/fabric8-ui

---
This message was auto-generated by https://gitcop.com

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -33,42 +35,63 @@ export const routes: Routes = [
...
+      context: ContextResolver
+    },
     loadChildren: './analyze/analyze.module#AnalyzeModule'
   },
 
   // Plan
   {
     path: ':entity/:space/plan',
+    resolve: {
+      context: ContextResolver
```
Comment (joshuawilson) [inline src/app/app-routing.module.ts]: What happens if you link directly to a submenu like the Planner Board menu. Does this resovler fire?

File: src/app/shared/context.service.ts [hunk 2]
```diff
@@ -92,17 +94,59 @@ export class ContextService implements Contexts {
...
+
+  get recent(): Observable<Context[]> {
+    return this._recent;
+  }
+
+  get current(): Observable<Context> {
+    return this._current;
+  }
+
+  get default(): Observable<Context> {
```
Comment (joshuawilson) [inline src/app/shared/context.service.ts]: "default" is a reserved word in TS

File: src/app/app-routing.module.ts (hunk 1 above)
Comment (pmuir) [inline src/app/app-routing.module.ts]: Yes, it fires exactly once for the node in the route tree it is attached to.

File: src/app/shared/context.service.ts (hunk 2 above)
Comment (pmuir) [inline src/app/shared/context.service.ts]: This isn't a change, it's just a moved method. See https://github.com/fabric8-ui/ngx-fabric8-wit/issues/34

**Pull request merged**